### ISSUE TITLE
PHP 7.x Support

### ISSUE THREAD

**Issue by povrazor** [krakjoe/apcu#129]
You're probably aware of this, but APCU 4.0.7 is currently incompatible with PHP 7.

```
 cc -D_GNU_SOURCE -I. -I/tmp/pear/temp/apcu -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-rootRa1uyd/apcu-4.0.7/include -I/tmp/pear/temp/pear-build-rootRa1uyd/apcu-4.0.7/main -I/tmp/pear/temp/apcu -I/usr/local/lsws/lsphp7/include/php -I/usr/local/lsws/lsphp7/include/php/main -I/usr/local/lsws/lsphp7/include/php/TSRM -I/usr/local/lsws/lsphp7/include/php/Zend -I/usr/local/lsws/lsphp7/include/php/ext -I/usr/local/lsws/lsphp7/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /tmp/pear/temp/apcu/apc.c  -fPIC -DPIC -o .libs/apc.o
In file included from /tmp/pear/temp/apcu/apc_sma.h:34:0,
                 from /tmp/pear/temp/apcu/apc_pool.h:33,
                 from /tmp/pear/temp/apcu/apc_cache.h:36,
                 from /tmp/pear/temp/apcu/apc.c:35:
/tmp/pear/temp/apcu/apc_sma_api.h:62:33: error: unknown type name ‘zend_uint’
 typedef void (*apc_sma_init_f) (zend_uint num, zend_ulong size, char *mask TSRMLS_DC);
                                 ^
/tmp/pear/temp/apcu/apc_sma_api.h:83:5: error: unknown type name ‘apc_sma_init_f’
     apc_sma_init_f init;                         /* init */
     ^
/tmp/pear/temp/apcu/apc_sma_api.h:103:5: error: unknown type name ‘zend_uint’
     zend_uint  num;                              /* number of segments */
     ^
/tmp/pear/temp/apcu/apc_sma_api.h:105:5: error: unknown type name ‘zend_uint’
     zend_uint  last;                             /* last segment */
     ^
/tmp/pear/temp/apcu/apc_sma_api.h:119:36: error: unknown type name ‘zend_uint’
                                    zend_uint num,
                                    ^
/tmp/pear/temp/apcu/apc_sma_api.h:224:52: error: unknown type name ‘zend_uint’
     PHP_APCU_API void apc_sma_api_func(name, init)(zend_uint num, zend_ulong size, char* mask TSRMLS_DC); \
                                                    ^
/tmp/pear/temp/apcu/apc_sma.h:37:1: note: in expansion of macro ‘apc_sma_api_decl’
 apc_sma_api_decl(apc_sma); /* }}} */
 ^
In file included from /tmp/pear/temp/apcu/apc_cache.h:40:0,
                 from /tmp/pear/temp/apcu/apc.c:35:
/tmp/pear/temp/apcu/apc_cache_api.h:45:5: error: unknown type name ‘zend_uint’
     zend_uint len;                /* length of data at str */
     ^
/tmp/pear/temp/apcu/apc_cache_api.h:55:5: error: unknown type name ‘zend_uint’
     zend_uint ttl;                /* the ttl on this specific entry */
     ^
/tmp/pear/temp/apcu/apc_cache_api.h:90:5: error: unknown type name ‘zend_ushort’
     zend_ushort state;               /* cache state */
     ^
/tmp/pear/temp/apcu/apc_cache_api.h:219:40: error: unknown type name ‘zend_uint’
                                        zend_uint keylen,
                                        ^
/tmp/pear/temp/apcu/apc_cache_api.h:221:40: error: unknown type name ‘zend_uint’
                                        const zend_uint ttl,
                                        ^
/tmp/pear/temp/apcu/apc_cache_api.h:228:41: error: unknown type name ‘zend_uint’
                                         zend_uint keylen,
                                         ^
/tmp/pear/temp/apcu/apc_cache_api.h:239:48: error: unknown type name ‘zend_uint’
                                                zend_uint keylen,
                                                ^
/tmp/pear/temp/apcu/apc_cache_api.h:248:40: error: unknown type name ‘zend_uint’
                                        zend_uint keylen,
                                        ^
/tmp/pear/temp/apcu/apc_cache_api.h:261:50: error: unknown type name ‘zend_uint’
                                                  zend_uint keylen,
                                                  ^
/tmp/pear/temp/apcu/apc_cache_api.h:269:41: error: unknown type name ‘zend_uint’
                                         zend_uint keylen TSRMLS_DC);
                                         ^
/tmp/pear/temp/apcu/apc_cache_api.h:304:54: error: unknown type name ‘zend_uint’
                                                      const zend_uint ttl TSRMLS_DC);
                                                      ^
/tmp/pear/temp/apcu/apc_cache_api.h:317:35: error: unknown type name ‘zend_uint’
                                   zend_uint keylen TSRMLS_DC);
                                   ^
/tmp/pear/temp/apcu/apc.c:46:43: fatal error: ext/standard/php_smart_str.h: No such file or directory
 #   include "ext/standard/php_smart_str.h"
                                           ^
compilation terminated.
```

**Comment by guilhermeblanco** [comment]
```bash
git checkout seven
make clean
phpize
make -j 4
sudo make install
```

**Comment by krakjoe** [comment]
This ^ ... Thanks ;)

**Comment by vincib** [comment]
Just a quick message to warn everyone : php7-compatible version of APCu is now in the MASTER branch of this repository, thanks !

**Comment by mikehaertl** [comment]
@krakjoe Since the release of 4.0.11 the command `pecl install apcu` now fails on PHP7 because it doesn't use 5.1.3. Instead pecl sees 4.0.11 as latest version, which is not compatible with PHP7. So I had to specify a version like

    pecl install apcu-5.1.3

But I don't quite like having static version constraints. Is there maybe a better way to fix this?

**Comment by Craeckie** [comment]
same here.
That workaround worked for me, but better fix would be appreciated.

**Comment by remicollet** [comment]
Better fix is to add support for REST-1.3 to pecl channel (as pear channel already have), so the pecl command will be aware of PHP dependency.

IMHO, nothing can be done in APCu.

=> https://pear.php.net/bugs/report.php?package=PEAR
